src/crimson/osd/pg.cc

```diff
@@ -1148,7 +1148,7 @@ seastar::future<std::optional<eversion_t>> PG::submit_error_log(
       return shard_services.send_to_osd(peer.osd,
                                         std::move(log_m),
                                         get_osdmap_epoch());
-    }).then([this, waiting_on, t=std::move(t), rep_tid] () mutable {
```

Comment:
Looks `do_transaction(t)` is the only user, so `t` doesn't need to be wrapped by `do_with()` to be moved there?

Reply:
Yeah, done, please take a look again.